src/asciidocParser.ts

```diff
@@ -16,7 +16,8 @@ export class AsciidocParser {
   private stylesdir = path.join(this.extPath, 'media')
 
   constructor (private readonly filename: string, private errorCollection: vscode.DiagnosticCollection = null) {
-    this.processor = asciidoctor()
+    this.filename = filename
```

Comment:
I put `this.filename` and `this.errorCollection` to avoid a useless constructor lint message -- is this OK?

Reply:
Last time I ran the linter, I didn't have a warning message.


```typescript
constructor (private readonly filename: string, private errorCollection: vscode.DiagnosticCollection = null) {
```

Using `private readonly filename: string` is valid TypeScript so, technically, we don't need to add `this.filename = filename`.

I wouldn't add extra code "just" to comply with the linter (which apparently does not recognize that this code is valid).In this case, I think it's better to ignore the linter warning.

Reply:
Fair enough, however I've done this elsewhere as well... :disappointed: 
I've added a commit to include https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-useless-constructor.md

Reply:
> I've added a commit to include https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-useless-constructor.md

Yes, that's the right way to fix it 👍